src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -111,7 +111,7 @@ internal static bool IsEnabled()
 
                 return await responseTask.ConfigureAwait(false);
             }
-            catch (TaskCanceledException)
+            catch (OperationCanceledException)
```

Comment:
Why have this catch block at all?  It just throws the exception.  Wouldn't removing this catch block end up doing the same thing?

Reply:
> Wouldn't removing this catch block end up doing the same thing?

No, then it would be caught and handled by the next catch block, which writes out additional failure output.  The alternative to this catch block would be to put a `where` filter on the below catch block.

Reply:
ok.  Would a `where` filter be faster than having this separate block?  If not, then this is fine.

Reply:
IIRC, a filter is a bit faster in the case of exception, but I'd need to re-measure.